canvas-gamification/canvas-gamification-ui
# Update problem sets components - Taiga UI

## Description
*warning lots of changes, sorry*
 - Update problem viewing component
 - Update submission viewing component
 - Update problem set component
 - Remove problem edit component
 - Update problem create component
   - Rename to problem-create-edit
   - Handle both editing and creating problems
   - Modify inner components to either take question data and populate form or create a new empty form
   - Update forms
   - Add default component to handle the same values between questions
 - Update json-editor component
   - Completely rewrite the whole process
   - The system is based on angular forms now
   - Different value types are determined by the name of the values given
 - Update tests
   - Tests that were not written
 - Update styling
 - Add pipes for templates
   - pipe to cast to form control
   - pipe to cast to from group
   - pipe to handle stringifying dropdown labels vs values
 - Add validator
   - form validator to ensure a field exists if another also exists

## Types of changes
What types of changes does your code introduce?
- [x] Refactor (only improve the code quality no change in functionality)

## Checklist:
- [x] Issue has been created
- [x] Every file touched is linted
- [x] Every file touched has tests that cover all the funcitonality with green coverage

## Issue
Closes #319 

## Tests
Update tests that failed from changes, add new tests for new/refactored features

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/37320754/147167845-71af20c5-f2f8-4a46-9c24-ae9bcf354bb5.png)
